validator_client/src/main.rs

```diff
@@ -119,13 +91,20 @@ fn main() {
     let poll_interval_millis = spec.seconds_per_slot * 1000 / 10; // 10% epoch time precision.
     info!(log, "Starting block producer service"; "polls_per_epoch" => spec.seconds_per_slot * 1000 / poll_interval_millis);
 
+    let keypairs = config
+        .fetch_keys()
+        .expect("Encountered an error while fetching saved keys.")
```

Comment:
All the errors you built, you just throw away here and panic. It would be simpler to just panic when the errors occurred in `fetch_keys` and have it not return a `Result<>`.